internal/provider/credential_provider_resource.go

```diff
@@ -307,6 +308,29 @@ func (r *credentialProviderResource) Schema(_ context.Context, _ resource.Schema
 						Description: "State for the OAuth Credential Provider.",
 						Computed:    true,
 					},
+					"lifetime": schema.Int64Attribute{
+						Description: "Lifetime multiplier of the OAuth Authorization Code credentials requested by the Credential Provider.",
+						Optional:    true,
```

Comment:
Let's make this required and default to 31536000 (1 year)

Reply:
Sure, I assume there should be a min check, which shouldn't be shorter than a day?

Reply:
That would be a great validation, yes